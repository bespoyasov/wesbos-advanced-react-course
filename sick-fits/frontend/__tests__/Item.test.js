import ItemComponent from '../components/Item'
import {shallow} from 'enzyme'
import toJSON from 'enzyme-to-json'

const fakeItem = {
  id: 'test id',
  title: 'test title',
  price: 5000,
  description: 'test description',
  image: 'test image.jpeg',
  largeImage: 'test large image.jpg',
}

describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})