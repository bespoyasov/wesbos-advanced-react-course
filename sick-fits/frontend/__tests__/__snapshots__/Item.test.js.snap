// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="test title"
    src="test image.jpeg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "test id",
          },
        }
      }
    >
      <a>
        test title
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    test description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "test id",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="test id"
    />
    <DeleteItem
      id="test id"
    >
      Delete
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
